Added GraphQL Queries, Basic Functionalities

Dutta-SD opened the pull request:
#2 
Added GraphQL Query to Fetch Repo Stats.

Dutta-SD committed:
Added GraphQL Queries, Basic Functionalities

Dutta-SD committed:
Added more documentation

Dutta-SD committed:
Added Detailed Printing

Dutta-SD committed:
Fixed Missing Bracket

smaranjitghose commented on Get_Repo_Details/config.py:
```diff
@@ -0,0 +1,30 @@
+
+BASE_URL = "https://api.github.com/graphql" # Github base url
+GITHUB_TOKEN = os.environ['GITHUB_TOKEN']  # Github token environment variable
+REPONAME, USERNAME = None, None
```
this is supposed to be entered as a single string

Dutta-SD replied:
Okay I made this into a command line tool using fire. Is it okay?

anushbhatia replied:
ya

smaranjitghose commented on Get_Repo_Details/config.py:
```diff
@@ -0,0 +1,30 @@
+BASE_URL = "https://api.github.com/graphql" # Github base url
+GITHUB_TOKEN = os.environ['GITHUB_TOKEN']  # Github token environment variable
+REPONAME, USERNAME = None, None
+
```
use fire library to take input

smaranjitghose commented on Get_Repo_Details/get_repo_details.py:
```diff
@@ -0,0 +1,49 @@
+
+# IMPORTS
+import json
+import requests
```
Have this in a single script. At this stage, there is no need for modularization

Dutta-SD replied:
Yes. Converted this into a single script

smaranjitghose commented on Get_Repo_Details/get_repo_details.py:
```diff
@@ -0,0 +1,49 @@
+  # Set names of Repo and gets data from API Query
+
+  config.REPONAME = input("Enter Repo Name as it appears : ")
+  config.USERNAME = input("Enter the owner of the repository as it appears : ")
```
All this needs to be taken using command line input

smaranjitghose commented on Get_Repo_Details/get_repo_details.py:
```diff
@@ -0,0 +1,49 @@
+  # Pretty display 
+  print("-" * 80)
+  print(f"Stats for Repository {config.REPONAME} (Owner : {config.USERNAME})")
+  print(f"Number of collaborators : {num_collaborators}")
```
Use pprint instead of print

Dutta-SD replied:
Should I print the whole returned json response using pprint ? Or print this individually  ?

anushbhatia replied:
print complete json

smaranjitghose commented:
Add a snapshot of the terminal output to your PR as well

smaranjitghose commented on Get_Repo_Details/get_repo_details.py:
```diff
@@ -0,0 +1,49 @@
+# Script to GET Repo details from user
+# @Author - Sandip Dutta
```
We will have all of the contributors and maintainers on the documentation page. This is not required in the scripts as of now

Dutta-SD replied:
Yes I will remove these lines from the script.

smaranjitghose commented on Get_Repo_Details/config.py:
```diff
@@ -0,0 +1,30 @@
+REPONAME, USERNAME = None, None
+
+def getQuery():
+  # Reuturns the Query String
```
Use DocStrings for describing functions

Dutta-SD committed:
Merged files, Added 1 docstring

1. Merged both files into one.    2. Added Docstring of one function

Dutta-SD commented:
Hi, This is the screenshot of the app.
![Git_Get_repo_Details](https://user-images.githubusercontent.com/57990650/105720244-3a484c00-5f49-11eb-857f-117910738748.png)

UPDATE - This was before adding and using fire module.

Dutta-SD committed:
Modified Code, Turned into module using fire

$ Modified code to take repo Name and user Name as command line inputs $ Taking input from CLI terminal using fire

anushbhatia commented:
Please use fire to get command line arguments as inputs properly. Your current script lack's it.

smaranjitghose commented on Get_Repo_Details/get_repo_details.py:
```diff
@@ -0,0 +1,90 @@
+        ... on Commit {{
+          history {{
+            totalCount
+          }}
```
This is a totally unnecessary function definition so initialize a string. remove it and  keep it as a simple variable

Dutta-SD replied:
Removed it

smaranjitghose commented on Get_Repo_Details/get_repo_details.py:
```diff
@@ -0,0 +1,90 @@
+  # Set names of Repo and gets data from API Query
+
+  # Sanity check
+  assert (repoName is not None and userName is not None), "Invalid Details"
```
Not at all necessary. A simple exit code does the work

smaranjitghose commented on Get_Repo_Details/get_repo_details.py:
```diff
@@ -0,0 +1,90 @@
+  # API Request
+  response = requests.post(
+    BASE_URL,
+    json={'query': getQuery(repoName, userName)},
```
# GitHub's API endpoint for  GraphQL
url = 'https://api.github.com/graphql'

# Headers for Authentication
headers = {'Authorization': "Token " + access_token}

# Making our API request
r = requests.post(url, json={'query': query}, headers=headers)

# Printing the Status Code to check for any error
print(r.status_code)

smaranjitghose commented on Get_Repo_Details/get_repo_details.py:
```diff
@@ -0,0 +1,90 @@
+  # Extract meaningful Information from json data
+  # As per Query Structure
+  num_collaborators = json_data['data']['repository']['collaborators']['totalCount']
+  num_pullRequests = json_data['data']['repository']['pullRequests']['totalCount']
```
Use simple names

pr_count
colab_count
star_count
commit_count

smaranjitghose commented on Get_Repo_Details/get_repo_details.py:
```diff
@@ -0,0 +1,90 @@
+
+
+
+if __name__ == '__main__':
```
Learn to use Fire in an effective way:

```python
import fire
import os
import json
import requests
import numpy as np
import pandas as pd


class GitKundli(object):

# Declare the GitHub access tokens globally

  def repo_info(self, username, reponame):
    """All the relevant code including print statements"""



if __name__ == '__main__':
  fire.Fire(Calculator)
```

Dutta-SD replied:
Yes sir, I modified the code like this. Please give me some feedback if it is okay now

Dutta-SD committed:
Modified i/o code, added printing

1. Added Pretty printing 2. Converted multiple functions into single class for better use with fire 3. Added Docstrings

Dutta-SD commented:
### This is how the CLI looks.
I added all changes. As specified, I displayed the json data.
![get_repo_sandip_dutta](https://user-images.githubusercontent.com/57990650/105880992-fed17e80-6029-11eb-8a7d-853307c108e9.png)
